Feature/seongho

rlatjdgh9612 opened the pull request:
## 1. 본 PR이 우리 팀의 웹 서비스 제품성에 어떠한 기여를 하였고, 사용자에게 어떠한 기대효과를 전달하는지 작성해주세요.
- 내 PR이 제품 내 어떠한 기능적인 배경/전후맥락 가운데 개발되었나요?

  (예시) 고객의 유입과 전환을 책임지는 회원 기능에서, 유저의 이탈을 방지하기 위해 input 창 바로 아래에서 실시간으로 적색의 경고 메세지를 전달 받습니다.

- 내 PR이 Merge 됨으로써 유저에게 전달되는 편익/기대효과는 무엇일까요?

  (예시) 유저는 실시간으로 스스로 오기입한 내용을 확인할 수 있기 때문에 잘못 계정 정보를 작성함으로써 발생될 수 있는 변수를 최소화할 수 있습니다. 이는 곧 유저 전환의 허들을 낮추어 줄 수 있습니다.



<br />

## 2. 이 브랜치에서 어떤 내용을 개발했는지 큰 제목과 그리고 상세 내역을 적어주세요.

(예시) 로그인 기능 추가

- (예시) email입력 `@`필요,password에 숫자로 8자이상 유효성 검사 추가
- (예시) 유효성 검사 통과시 버튼 컬러 변경 및 버튼 활성화

<br />

## 3. 개발한 화면을 캡쳐해서 첨부 해 주세요. (drag & drop 또는 첨부파일 추가)
<br />

-

<br />

## 4. 이 브랜치에서 개발하면서 느꼇던 개발 성장포인트를 적어주세요.

- (예시) 새로운 배열메서드 filter를 쓸 수 있게됐습니다.
- (예시) 상수데이터와 map 메서드로 반복되는 UI를 그릴 수 있게 됐습니다.
- (예시) fetch함수를 어떻게 쓰는지 알게됐습니다.
- (예시) 계산된 속성명 (computed property)을 쓰긴했는데 어떻게 돌아가는지 이해가 잘 안되서
  블로그에 정리해봐야겠습니다.
- (예시) 화살표 함수(arrow-function) 어떤 코드를 언제 생략할 수있는지 완벽 이해했습니다.
- (예시) 구조분해할당(Destructuring assignment)을 써보니 가독성도, 유지보수도 편할것같습니다.
  <br />

rlatjdgh9612 committed:
git 브랜치 생성
git log
git branch
clear

rlatjdgh9612 committed:
git 브랜치 생성

rlatjdgh9612 committed:
git 브랜치 생성 완료

rlatjdgh9612 committed:
hello

5yoonl commented on src/Router.js:
```diff
@@ -1,14 +1,18 @@
-        <Route path="/kimcode-main" element={<KimCodeMain />} />
+        <Route path="/login" element={<Login />} />
+        <Route path="/register" element={<Register />} />
+        <Route path="/signup_compliete" element={<Signup_Compliete />} />
```
path값을 작성할 때는 a-b 형식으로 작성합니다!  _ 대신  -를 사용해주세요

rlatjdgh9612 replied:
넵! 수정 완료하였습니다:)

5yoonl commented on src/pages/SeongHoKim/Login/Login.jsx:
```diff
@@ -0,0 +1,90 @@
+  };
+
+  return (
+    <div className="login_page">
```
전반적으로 className은 camelCase사용해주세요

rlatjdgh9612 replied:
현재 모든 Page className들을 camelCase로 변경하였습니다.

5yoonl commented on src/pages/SeongHoKim/Login/Login.jsx:
```diff
@@ -0,0 +1,90 @@
+    <div className="login_page">
+      <div className="logo_frame">
+        <img className="big_logo" src={logo1} alt="위코드 로고1"></img>
+        <img className="small_logo" src={logo2} alt="위코드 로고2"></img>
```
content가 없는 태그에 대해서는 self-closing 사용해주세요
```suggestion
        <img className="big_logo" src={logo1} alt="위코드 로고1" />
        <img className="small_logo" src={logo2} alt="위코드 로고2" />
```

rlatjdgh9612 replied:
넵. self-closing 으로 수정하였습니다.

5yoonl commented on src/pages/SeongHoKim/Login/Login.scss:
```diff
@@ -0,0 +1,115 @@
+    justify-content: center;
+    margin-top: 400px;
+
+    .email,
+    .password {
+      width: 450px;
+      padding: 15px;
+      margin-top: 7px;
+      border: 1px solid $border_color;
+      border-radius: 5px;
+
+      &::placeholder {
+        color: #999;
+      }
+    }
```
두가지 className이  모두 동일한 속성을 지니고있습니다 이럴땐 HTML영역에서 같은 className을 주는것이 더 효과적입니다

5yoonl commented on src/pages/SeongHoKim/Register/Register.jsx:
```diff
@@ -0,0 +1,189 @@
+          <option value="1996년">1996년</option>
+          <option value="1997년">1997년</option>
+          <option value="1998년">1998년</option>
+          <option value="1999년">1999년</option>
```
year month day 모두 각각의 같은UI가 비효율적인 방식으로 반복되고있습니다.  상수데이터를  구성하고 map을 활용한방식으로 수정해보세요

rlatjdgh9612 replied:
넵 map 메서드 이용해서 기능 수정해서 적용 완료했습니다.

5yoonl commented on src/pages/SeongHoKim/Register/Signup_Compliete.jsx:
```diff
@@ -0,0 +1,42 @@
+import { Navigate, useNavigate } from 'react-router-dom';
+
+const Signup_Compliete = () => {
+  const moveLogin = useNavigate();
```
useNavigate hook에서 반환되는 함수가 login페이지로만 이동하기위한 기능을  제공하는것이 아니기 때문에 적절하지 못한 이름의 변수인것 같습니다. 일반적으로 사용하는 navigate 변수를 사용해주세요

5yoonl reviewed (commented):
수고하셨습니다 성호님! 전반적으로 camelCase사용, 및 과도한 주석사용, css 코드 사이 의문의 빈 공백 등등 모든부분에대해 코멘트를 남기지는  않았지만 전체적으로 수정해주세요!

rlatjdgh9612 committed:
1. birthday year, month, day UI list들이 비효율적으로 동작하여 map 메서드로 수정하였습니다.
2. phone_number UI list map 메서드로 수정하였습니다.

rlatjdgh9612 committed:
1. 전화번호, 생일 List들을  Map 메서드로 수정하기 - 완료
2. Router path값 -으로 수정하기 - 완료
3. content가 없는 이미지 태그를 self-closing으로 수정하기 - 완료
4. login, signup, signup-compliete className을 모두 camelCase로  수정하기 - 완료

rlatjdgh9612 committed:
1.self-closing 미적용된 img 태그 self-closing 적용하기
